fn.go

```diff
@@ -121,24 +122,44 @@ func (f *Function) RunFunction(_ context.Context, req *fnv1.RunFunctionRequest)
 
 // Build requirements takes input and outputs an array of external resoruce requirements to request
 // from Crossplane's external resource API.
-func buildRequirements(_ *v1alpha1.Input, xr *resource.Composite) *fnv1.Requirements {
-	extraResources := make(map[string]*fnv1.ResourceSelector)
+func buildRequirements(_ *v1alpha1.Input, xr *resource.Composite, context *structpb.Struct) *fnv1.Requirements {
 	spec := xr.Resource.Object["spec"].(map[string]any)
+
+	env := context.GetFields()["apiextensions.crossplane.io/environment"].GetStructValue()
+	if env == nil {
+		return &fnv1.Requirements{}
+	}
+
+	observedTenant := env.GetFields()["tenantName"].GetStringValue()
+	observedAccount := spec["accountRef"].(map[string]any)["name"].(string)
+
+	extraResources := make(map[string]*fnv1.ResourceSelector)
 	permissions, ok := spec["permissions"].([]any)
 	if ok {
 		for _, permission := range permissions {
 			for _, principal := range permission.(map[string]any)["principals"].([]any) {
-				user, ok := principal.(map[string]any)["user"]
-				if ok {
-					extraResources[user.(string)] = &fnv1.ResourceSelector{
+				principal := principal.(map[string]any)
```

Comment:
extract `user`, `tenant` and `account` in a dedicated function? (You'd either get all 3 or an error)

Reply:
I think you should be allow to get only `user`

Reply:
What does not make sense is to only specify `tenant` and `user` - not `account`